### PR TITLE
Support Sensu go 5.12

### PR DESCRIPTION
**Pull request by treydock**
# Pull Request Checklist

## Description

* Add `runtime_assets` property to `sensu_hook`.
* Add `builds` property to `sensu_asset`
* Support removing `sensu_asset` types
* Add `sensu_oidc_auth` type

## Motivation and Context

These are changes that were added in Sensu Go 5.12

### CODE REVIEW

**Comment by treydock** [comment]
I marked WIP as 5.12 isn't out yet but am able to test using CI builds. Also the official changelog hasn't been pushed so maybe more changes to come.

**Comment by treydock** [comment]
Added `sensu_oidc_auth` type based on pending documentation updates for 5.12. Work on CI builds.

File: lib/puppet/provider/sensu_oidc_auth/sensuctl.rb [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    auth_types = sensuctl_auth_types()
+    data.each do |d|
+      auth = {}
+      auth[:ensure] = :present
+      auth[:name] = d['metadata']['name']
+      if auth_types[auth[:name]] != 'OIDC'
+        next
+      end
+      #auth[:client_id] = d['client_id']
+      #auth[:client_secret] = d['client_secret']
```
Comment (ghoneycutt) [inline lib/puppet/provider/sensu_oidc_auth/sensuctl.rb]: why the commented out code?

Comment (treydock) [reply lib/puppet/provider/sensu_oidc_auth/sensuctl.rb]: That was leftover from development, I removed commented out code.

**Pull request merged**